packages/@aws-cdk/aws-iam/lib/policy-statement.ts

```diff
@@ -1,25 +1,65 @@
 import * as cdk from '@aws-cdk/core';
-import { AccountPrincipal, AccountRootPrincipal, Anyone, ArnPrincipal, CanonicalUserPrincipal,
-  FederatedPrincipal, IPrincipal, ServicePrincipal, ServicePrincipalOpts } from './principals';
+import {
+  AccountPrincipal, AccountRootPrincipal, Anyone, ArnPrincipal, CanonicalUserPrincipal, FederatedPrincipal, IPrincipal,
+  ServicePrincipal, ServicePrincipalOpts
+} from './principals';
 import { mergePrincipal } from './util';
 
 /**
  * Represents a statement in an IAM policy document.
  */
 export class PolicyStatement {
+
+  /**
+   * Creates a new PolicyStatement based on the object provided.
+   * This will accept an object created from the `.toJSON()` call
+   * @param obj the PolicyStatement in object form.
+   */
+  public static fromJSON(obj: any): PolicyStatement {
+    const statement = new PolicyStatement({
+      actions: obj.Action,
+      resources: obj.Resource,
+      conditions: obj.Condition,
+      effect: obj.Effect,
+      notActions: typeof (obj.NotAction) === "string" ? [obj.NotAction] : obj.NotAction,
+      notResources: typeof (obj.NotResource) === "string" ? [obj.NotResource] : obj.NotResource
+    });
+
+    if (obj.Principal && obj.Principal.AWS) {
```

Comment:
Instead of all these cases we can add `principals` to `PolicyStatementProps` and just pass `obj.Principal` through like the other fields, no?